src/plugins/elasticsearch/index.js

```diff
@@ -0,0 +1,149 @@
+const elasticsearch = require('elasticsearch');
+const accessLogsIndexConfig = require('./access-logs_index.json');
+const config = require('../../constants');
+
+const accessLogsIndex = 'access-watch-access-logs';
+
+const getIndexSuffix = date =>
+  [date.getMonth() + 1, date.getDate(), date.getFullYear()].join('-');
+
+const generateIndex = date => `${accessLogsIndex}-${getIndexSuffix(date)}`;
+
+const generateCurrentIndex = () => generateIndex(new Date());
+const getIndexDate = index =>
+  index.slice(accessLogsIndex.length + 1).replace(/-/g, '/');
+
+const indexAccessLog = client => log =>
+  client.index({
+    index: generateCurrentIndex(),
+    type: 'access-log',
+    routing: log.getIn(['address', 'value']),
+    body: log.toJS(),
+  });
+
+const indexesGc = client => () => {
+  client.indices.get({ index: '*' }).then(indices => {
+    Object.keys(indices)
+      .filter(i => i.indexOf(accessLogsIndex) !== -1)
+      .forEach(index => {
+        const indexDate = new Date(getIndexDate(index));
+        const gcDate = new Date(
+          new Date().getTime() -
+            config.elasticsearch.retention * 24 * 3600 * 1000
+        );
+        if (indexDate.getTime() < gcDate.getTime()) {
+          client.indices.delete({ index });
+        }
+      });
+  });
+};
+
+const logsSearchArguments = {
+  address: value => ({ address: { value } }),
+  identity_type: type => ({ identity: { type } }),
+  request_method: method => ({ request: { method } }),
+  reputation_status: status => ({ reputation: { status } }),
+  response_status: status => ({ response: { status } }),
+  robot: id => ({ robot: { id } }),
+};
```

Comment:
Is it to accomodate previous parameters we have?

Maybe can we imagine a new style of query parameters that works by default? Like address.value, robot.id, request.method, etc ...

Reply:
Hmmm.
I agree and disagree.
If we do go for `address.value` I would be tempted to accept anything, which gives the power to the frontend to basically query anything elasticsearch wise, which is nice, but which also means we do not control anymore what it can query.
We can put back the control with an array of authorized query terms, but it means we are introducing query style specifing to elasticsearch, I am not necessarily against it, it is just it might make people think you can do more than you actually can in the end.